src/snapping/index.js

```diff
@@ -629,34 +573,7 @@ class Snapping {
   _updateSnapLayers() {
     if (!this.snappingEnabled) return;
 
-    setTimeout(() => {
-      const newLayers = this._snappableLayers();
-
-      this.bufferLayers
-        .filter((l) => !newLayers.includes(l))
-        .forEach((l) => this._removeSnapBuffer(l));
-
-      newLayers
-        .filter((l) => !this.bufferLayers.includes(l))
-        .forEach((l) => this._addSnapBuffer(l));
-
-      newLayers
-        .filter((l) => this.bufferLayers.includes(l))
-        .forEach((l) => {
-          const bufferLayerId = getBufferLayerId(l);
-
-          if (this.map.getLayer(bufferLayerId)) {
-            const parentFilter = this.map.getLayer(l).filter;
-
-            const bufferFilter = parentFilter.filter(
-              (filt) => !(filt instanceof Array) || filt[0] !== "!="
-            );
-
-            this.map.setFilter(bufferLayerId, bufferFilter);
-          }
-        });
-      this.bufferLayers = newLayers;
-    });
+    this.snapLayers = this._snappableLayers();
```

Comment:
we still keep an array of layers that can be snapped to so we are not calling `this.map.getStyle().layers.filter...` on every mouse move. This may be needless but given this can be run 20 times per second, seems better to keep it "cached"